compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/res/resource.uikit.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.res
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.ImageBitmap
+
+import platform.Foundation.*
+import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.usePinned
+import platform.Foundation.NSBundle
+import platform.Foundation.NSData
+import platform.posix.memcpy
+import androidx.compose.ui.graphics.toComposeImageBitmap
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.skia.Image
+
+fun interface ByteSource { //todo move to commonMain
+    suspend fun readBytes(): ByteArray //todo in future use streaming
+}
+
+fun interface ResourceProvider<T> { //todo move to commonMain
+    suspend fun provide(): T
+}
+
+@Composable
+fun <T> ResourceProvider<T>.rememberBlocking() = remember { //todo move to commonMain
```

Comment:
I think it's necessary to provide some key to "remember" call: `remember(key) { ... }`

Here is a use-case:
```kotlin
val path = if (someCondition) "img.png" else "abc.png"
Image(bitmap = resource(path).asImageBitmap().rememberBlocking())
```

If no key to remember provided, then `rememberBlocking` will be returning the very first value regardless of `path` value.

At first I thought that we can use `this` (ResourceProvider) instance as a key, but there will be a new instance for every new invocation. - leading to new calculations every time.

Maybe we need to use a usual interface for ResourceProvider and ByteSource and provide a correct `equals` implementation (so then we can use `this` as a key for remember). 
Or we can let users provide the key: `rememberBlocking(key = path)` - key might be optional.